### ISSUE TITLE
Pulse highlight row on interface refresh / change

### ISSUE THREAD

**Issue by asfaltboy** [timbrel/GitSavvy#552]
## Reasoning

In relation to PR #549 , I believe it would help if we could pulse the line the caret is on, after a refresh, to remind users to mind their action if the line now references another commit/branch/etc.

I just discovered and enabled the `"highlight_line": true` option in sublime and it really helps to see what row the caret is on clearly. But users may still have it off, or be too used to it to notice a change has occurred.

## Implementation suggestion

[This POC plugin][1] shows how one can "pulse" sublime's background color; I think we can do something similar to `lineHighlight` color (only with much bigger/quicker steps), something in the line of:

1. Backup the color of `lineHighlight` and set it to same as `background` color.
2. Enable `highlight_line` setting; the line highlight is still invisible at this point.
3. Quickly fade the `lineHighlight` color to saved original color and straiught back to `background` color, creating a pulsing highlight effect.
4. Disable the `highlight_line`.

If the user already has `highlight_line` enabled, we could pick a "brighter" target color instead, and fade into it and back (without toggling `highlight_line` on/off).

[1]: https://github.com/cryzed/Sublime-Text-3-Pulse-Plugin/blob/master/plugin.py#L123-L136

**Comment by divmain** [comment]
This is a fantastic idea!

**Comment by kaste** [comment]
I've set `highlight_line: true` for all of our fancy, line-based views.  
(But of course, many users will have them on on *all* their views.)

I don't want pulsing at all rather that the cursor is always at a good
position.  Esp. on a background refresh the cursors should just stay 
as they were.  Otherwise we want to be fast and don't fade out 
disappearing content.  

I also use `add_regions` to highlight areas of interest e.g. in the 
graph view. 

Closing therfore.
